File: PetStoreClient/petStore/src/app/models/order-item.model.ts
export class OrderItem{
    
    orderItemId: number | null;
    orderId: number | null;
    productId: number | null;
    price: number | null;
    quantity: number | null;

    productName: string | null;

    constructor(orderItemId?: number, orderId?: number, productId?: number, price?: number, quantity?: number, productName?: string){
        this.orderItemId = orderItemId || null;
        this.orderId = orderId || null;
        this.productId = productId || null;
        this.price = price || null;
        this.quantity = quantity || null;
        this.productName = productName || null;
    }

}

File: PetStore/Models/OrderItemModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace PetStore.Models
{
    public class OrderItemModel
    {
        public int? OrderItemId { get; set; }
        public int? OrderId { get; set; }

        public int ProductId { get; set; }
        public string ProductName { get; set; }

        public decimal? Price { get; set; }
        public int? Quantity { get; set; }
    }
}
